pkg/karmadactl/cmdinit/kubernetes/statefulset.go

```diff
@@ -351,14 +351,22 @@ func (i *CommandInitOption) makeETCDStatefulSet() *appsv1.StatefulSet {
 		Spec: podSpec,
 	}
 
+	var stsPvcRetentionPolicy *appsv1.StatefulSetPersistentVolumeClaimRetentionPolicy
+	if i.EtcdStsPvcDeletePolicy == string(appsv1.DeletePersistentVolumeClaimRetentionPolicyType) {
+		stsPvcRetentionPolicy = &appsv1.StatefulSetPersistentVolumeClaimRetentionPolicy{
+			WhenDeleted: appsv1.DeletePersistentVolumeClaimRetentionPolicyType,
+			WhenScaled:  appsv1.RetainPersistentVolumeClaimRetentionPolicyType,
```

Comment:
I have a question, I don't understand why this is set up.

Reply:
> why this is set up.

Default policy is appsv1.RetainPersistentVolumeClaimRetentionPolicyType; if set flag, WhenDeleted value is appsv1.DeletePersistentVolumeClaimRetentionPolicyType